src/index.ts

```diff
@@ -402,3 +451,75 @@ extendEnvironment(hre => {
         }
     };
 });
+
+task("starknet-verify", "Verifies the contract in the Starknet network.")
+    .addOptionalParam("starknetNetwork", "The network version to be used (e.g. alpha)")
+    .addParam("path", `The path of the cairo contract (e.g. contracts/conract.cairo)`)
+    .addParam("address", `The address where the contract is deployed`)
+    .setAction(async (args, hre) => {
+        var voyagerUrl = VOYAGER_GOERLI_CONTRACT_API_URL;
+        
+        if(args.starknetNetwork && args.starknetNetwork !== ALPHA){
+            if(args.starknetNetwork === ALPHA_MAINNET)
+                voyagerUrl = VOYAGER_MAINNET_CONTRACT_API_URL;
+            else{
+                const msg = `Unknown starknet-network provided: ${args.starknetNetwork}`;
+                throw new HardhatPluginError(PLUGIN_NAME, msg);
+            }
+        }
+        voyagerUrl+=args.address + "/code";
+        var isVerified = false;
+        await axios.get(voyagerUrl,{
+            headers: {
+              'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8',
+              'Content-Type': 'application/json'
+            }
+           })
+        .then(response => {
+            if(response.data.contract != null && response.data.contract.length > 0){
+                console.log(`Contract at address ${args.address} has already been verified`);
+                isVerified = true;
+            }
+        })
+        .catch(error => {
+            const msg = `Something went wrong when trying to verify the code at address ${args.address}`;
+            console.log(error);
+            throw new HardhatPluginError(PLUGIN_NAME, msg);
+        });
+        
+        if(isVerified)
```

Comment:
Should the user be notified about contract already having been verified?
Also indent the return statement and put it in curly braces.